From e830204abc64786b3eeed62322f2e3b79c2857f3 Mon Sep 17 00:00:00 2001
From: Jonathon Penix <jpenix@qti.qualcomm.com>
Date: Tue, 6 Oct 2026 13:55:44 -0700
Subject: [PATCH] [libc++] Reorder link flags in exception_guard.odr.sh.cpp

Moving `%{link_flags}` after the test object files resolves undefined reference
errors to ex: `strcmp` when using linkers that are more strict about the
command line order of objects.
---
 libcxx/test/libcxx-03/utilities/exception_guard.odr.sh.cpp | 2 +-
 libcxx/test/libcxx/utilities/exception_guard.odr.sh.cpp    | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/libcxx/test/libcxx-03/utilities/exception_guard.odr.sh.cpp b/libcxx/test/libcxx-03/utilities/exception_guard.odr.sh.cpp
index 92ec3e97cabe..4bd8f3f9cbd7 100644
--- a/libcxx/test/libcxx-03/utilities/exception_guard.odr.sh.cpp
+++ b/libcxx/test/libcxx-03/utilities/exception_guard.odr.sh.cpp
@@ -14,7 +14,7 @@
 
 // RUN: %{cxx} %s %{flags} %{compile_flags} -c -o %t.except.o   -O1 -fexceptions
 // RUN: %{cxx} %s %{flags} %{compile_flags} -c -o %t.noexcept.o -O1 -fno-exceptions
-// RUN: %{cxx} %{flags} %{link_flags} -o %t.exe %t.except.o %t.noexcept.o
+// RUN: %{cxx} %{flags} -o %t.exe %t.except.o %t.noexcept.o %{link_flags}
 // RUN: %{run}
 
 #include <__cxx03/__utility/exception_guard.h>
diff --git a/libcxx/test/libcxx/utilities/exception_guard.odr.sh.cpp b/libcxx/test/libcxx/utilities/exception_guard.odr.sh.cpp
index f6b8a29d3682..a1a4ad84e61d 100644
--- a/libcxx/test/libcxx/utilities/exception_guard.odr.sh.cpp
+++ b/libcxx/test/libcxx/utilities/exception_guard.odr.sh.cpp
@@ -14,7 +14,7 @@
 
 // RUN: %{cxx} %s %{flags} %{compile_flags} -c -o %t.except.o   -O1 -fexceptions
 // RUN: %{cxx} %s %{flags} %{compile_flags} -c -o %t.noexcept.o -O1 -fno-exceptions
-// RUN: %{cxx} %{flags} %{link_flags} -o %t.exe %t.except.o %t.noexcept.o
+// RUN: %{cxx} %{flags} -o %t.exe %t.except.o %t.noexcept.o %{link_flags}
 // RUN: %{run}
 
 #include <__utility/exception_guard.h>
-- 
2.43.0

